doc/ondemand.md

```diff
@@ -50,30 +31,42 @@ for (auto tweet : doc["statuses"]) {
+* **Streaming (\*):** It avoids preparsing values, keeping the memory usage and the latency down. 
+* **Forward-Only:** To prevent reiteration of the same values and to keep the number of variables down (literally), only a single index is maintained and everything uses it (even if you have nested for loops). This means when you are going through an array of arrays, for example, that the inner array loop will advance the index to the next comma, and the array can just pick it up and look at it.
+* **Natural Iteration:** A JSON array or object can be iterated with a normal C++ for loop. Nested arrays and objects are supported by nested for loops.
+* **Use-Specific Parsing:** Parsing is always specific to the type required by the programmer. For example, if the programmer asks for an unsigned integer, we just start parsing digits. If there were no digits, we toss an error. There are even different parsers for `double`, `uint64_t` and `int64_t` values. This use-specific parsing avoids the branchiness of a generic "type switch," and makes the code more inlineable and compact.
+* **Validate What You Use:** On Demand deliberately validates the values you use and the structure leading to it, but nothing else. The goal is a guarantee that the value you asked for is the correct one and is not malformed: there must be no confusion over whether you got the right value.
+
+
+To understand why On Demand is different, it is helpful to review the major
 approaches to parsing and parser APIs in use today.
 
-### Generic DOM Parsers
+### DOM Parsers
 
-Many of the most usable, popular JSON APIs deserialize into a **DOM**: an intermediate tree of
-objects, arrays and values. The resulting API lets you refer to each array or object separately,
-using familiar techniques like iteration (`for (auto value : array)`) or indexing (`object["key"]`).
-In some cases the values are even deserialized straight into familiar C++ constructs like vector and
-map.
+Many of the most usable, popular JSON APIs (including simdjson) deserialize into a **DOM**: an intermediate tree of
+objects, arrays and values.  In this model, we convert the input data all at once into a tree-like structure (the DOM). 
+The DOM is then accessed by the programmer like any other in-memory data structure.  The resulting API let 
+you refer to each array or object separately, using familiar techniques like iteration (`for (auto value : array)`) 
+or indexing (`object["key"]`). In some cases, the values are even deserialized directly into familiar C++ constructs like vectors and
+maps.
 
-This model is dead simple to use, since it talks in terms of *data types* instead of JSON. It's
-often easy enough that many users use the deserialized JSON as-is instead of deserializing into
-their own custom structs, saving a ton of development work.
+ The DOM approach is conceptually simple and "programmer friendly". Using the
```

Comment:
double space

Reply:
I am removing all of the double spaces throughout.